src/Kiota.Builder/Writers/TypeScript/CodeFunctionWriter.cs

```diff
@@ -54,14 +54,173 @@ public override void WriteCodeElement(CodeFunction codeElement, LanguageWriter w
+        foreach (var mappedType in composedType.GetNonPrimitiveTypes(IsComposedPrimitive).ToArray())
+        {
+            var mappedTypeName = mappedType.Name.ToFirstCharacterUpperCase();
+            writer.WriteLine($"...{GetFunctionName(codeElement, mappedTypeName, CodeMethodKind.Deserializer)}({composedParam.Name.ToFirstCharacterLowerCase()} as {GetTypescriptTypeString(mappedType, codeElement, includeCollectionInformation: false, inlineComposedTypeString: true)}),");
+        }
+        writer.CloseBlock();
+    }
+
+    private void WriteComposedTypeSerializer(CodeFunction codeElement, LanguageWriter writer, CodeParameter composedParam)
+    {
+        if (composedParam is null || GetOriginalComposedType(composedParam) is not { } composedType) return;
+
+        if (composedType.IsComposedOfPrimitives(IsComposedPrimitive))
+        {
+            WriteSerializationFunctionForTypeComposedOfPrimitives(composedType, composedParam, codeElement, writer);
+            return;
+        }
+
+        if (composedType is CodeIntersectionType)
+        {
+            WriteSerializationFunctionForCodeIntersectionType(composedType, composedParam, codeElement, writer);
+            return;
+        }
+
+        WriteSerializationFunctionForCodeUnionTypes(composedType, composedParam, codeElement, writer);
+    }
+
+    private void WriteSerializationFunctionForCodeIntersectionType(CodeComposedTypeBase composedType, CodeParameter composedParam, CodeFunction method, LanguageWriter writer)
+    {
+        // Serialization/Deserialization functions can be called for object types only
+        foreach (var mappedType in composedType.GetNonPrimitiveTypes(IsComposedPrimitive).ToArray())
+        {
+            var mappedTypeName = mappedType.Name.ToFirstCharacterUpperCase();
+            writer.WriteLine($"{GetFunctionName(method, mappedTypeName, CodeMethodKind.Serializer)}(writer, {composedParam.Name.ToFirstCharacterLowerCase()} as {GetTypescriptTypeString(mappedType, method, includeCollectionInformation: false, inlineComposedTypeString: true)});");
+        }
+    }
+
+    public static bool IsComposedOfCollectionOfObjects(CodeComposedTypeBase composedType)
+    {
+        ArgumentNullException.ThrowIfNull(composedType);
+        // Get the objects
+        var composedTypeObjects = composedType.Types.Where(x => !IsPrimitiveType(GetTypescriptTypeString(x, composedType)));
+        // Get the collections count
+        return composedTypeObjects.Any(x => x.IsCollection);
+    }
+
+    public static bool IsComposedOfNonCollectionOfObjects(CodeComposedTypeBase composedType)
+    {
+        ArgumentNullException.ThrowIfNull(composedType);
+        // Get the objects
+        var composedTypeObjects = composedType.Types.Where(x => !IsPrimitiveType(GetTypescriptTypeString(x, composedType)));
+        // Get the non collections count
+        return composedTypeObjects.Any(x => !x.IsCollection);
+    }
```

Comment:
This is effectively the inverse of the precedent one, refactor to avoid duplication